solidity/contracts/bank/Bank.sol

```diff
@@ -138,8 +151,10 @@ contract Bank is Ownable {
         );
     }
 
-    /// @notice Atomically increases the balance allowance granted to `spender`
-    ///         by the caller by the given `addedValue`.
+    /// @notice Atomically increases the caller's balance allowance granted to
```

Comment:
What are we using the word "atomically" to mean here? In something concurrent, I would imagine anything atomic to happen threadsafe, either all or nothing. Is that the same here?

Reply:
Let's say that balance owner A approved address B to use 100 units of A's balance in the Bank calling `Bank.approveBalance(B, 100)`.

After some time, A changed their mind and decided to change the allowance to 120 units. 

If A makes it by calling `Bank.approveBalance(B, 120)`, malicious B can front-run this transaction and first spend 100 units from the first approval, and then spend 120 units from the second approval, spending 220 total. Definitely not what was A's intention:
```
A: Bank.approveBalance(B, 100)
B: Bank.transferBalanceFrom(A, B, 100)
A: A: Bank.approveBalance(B, 120)
B: Bank.transferBalanceFrom(A, B, 120)
```

If A uses `Bank.increaseBalanceAllowance(B, 20)` instead, malicious B can spend no more than 120. This is why we say it happened *atomically*. It's similar to a database transaction where nothing can happen in between.